File: examples/durability/reminder-escalation/java/ReminderEscalationAgent.java
package ai.axme.examples.durability;

import ai.axme.examples.SseHelper;
import dev.axme.sdk.*;
import java.util.*;

/** Pre-Approval Validator — durability/reminder-escalation (Java). */
public class ReminderEscalationAgent {
    static final String BASE = SseHelper.env("AXME_BASE_URL", "https://api.cloud.axme.ai");
    static final String KEY = System.getenv("AXME_API_KEY");
    static final String ADDR = SseHelper.env("AXME_AGENT_ADDRESS", "pre-approval-validator");
    static final Set<String> TYPES = Set.of("config_update", "dependency_update", "rollback", "hotfix");
    static final Set<String> RISKS = Set.of("low", "medium", "high", "critical");

    public static void main(String[] args) throws Exception {
        if (KEY == null || KEY.isEmpty()) { System.err.println("AXME_API_KEY is required"); System.exit(1); }
        AxmeClient client = new AxmeClient(new AxmeClientConfig(BASE, KEY));
        System.out.printf("pre-approval-validator starting  address=%s%n", ADDR);
        int since = 0;
        while (true) {
            try {
                for (Map<String, Object> d : SseHelper.pollAgentStream(BASE, KEY, ADDR, since, 15)) {
                    String id = SseHelper.str(d, "intent_id"); Number seq = (Number) d.get("seq");
                    if (seq != null) since = Math.max(since, seq.intValue()); if (id.isEmpty()) continue;
                    Map<String, Object> intent = SseHelper.unwrapIntent(client.getIntent(id, RequestOptions.none()));
                    if (!SseHelper.isActionable(intent)) continue;
                    Map<String, Object> p = SseHelper.effectivePayload(intent);

                    String cid = SseHelper.str(p, "change_id", "?"), ct = SseHelper.str(p, "change_type");
                    String rl = SseHelper.str(p, "risk_level"), svc = SseHelper.str(p, "service");
                    boolean valid = true; String reason;
                    if (svc.isEmpty()) { valid = false; reason = cid + ": service is required"; }
                    else if (!TYPES.contains(ct)) { valid = false; reason = cid + ": change_type not allowed"; }
                    else if (!RISKS.contains(rl)) { valid = false; reason = cid + ": risk_level not valid"; }
                    else { reason = cid + ": pre-validation passed for " + svc; }

                    client.resumeIntent(id, Map.of("action", valid ? "complete" : "fail", "valid", valid, "reason", reason),
                            SseHelper.withOwner(ADDR));
                    System.out.printf("resumed %s (valid=%s)%n", id, valid);
                }
            } catch (Exception e) { Thread.sleep(2000); }
        }
    }
}

File: examples/dotnet/Durability/ReminderEscalationAgent.cs
using System.Text.Json.Nodes;
using Axme.Sdk;
using AxmeExamples;

var baseUrl = SseHelper.Env("AXME_BASE_URL", "https://api.cloud.axme.ai");
var apiKey = SseHelper.Env("AXME_API_KEY");
var addr = SseHelper.Env("AXME_AGENT_ADDRESS", "pre-approval-validator");
if (string.IsNullOrEmpty(apiKey)) { Console.Error.WriteLine("AXME_API_KEY is required"); return; }

var client = new AxmeClient(new AxmeClientConfig { ApiKey = apiKey, BaseUrl = baseUrl });
Console.WriteLine($"pre-approval-validator starting  address={addr}");
string[] types = ["config_update", "dependency_update", "rollback", "hotfix"];
string[] risks = ["low", "medium", "high", "critical"];
var since = 0;

while (true)
{
    try
    {
        foreach (var d in await SseHelper.PollAgentStreamAsync(baseUrl, apiKey, addr, since))
        {
            var id = SseHelper.Str(d, "intent_id");
            if (d["seq"]?.GetValue<int>() is int seq) since = Math.Max(since, seq);
            if (string.IsNullOrEmpty(id)) continue;
            var intent = SseHelper.UnwrapIntent(await client.GetIntentAsync(id));
            if (!SseHelper.IsActionable(intent)) continue;
            var p = SseHelper.EffectivePayload(intent);

            var cid = SseHelper.Str(p, "change_id", "?"); var ct = SseHelper.Str(p, "change_type");
            var rl = SseHelper.Str(p, "risk_level"); var svc = SseHelper.Str(p, "service");
            bool valid; string reason;
            if (string.IsNullOrEmpty(svc)) { valid = false; reason = $"{cid}: service required"; }
            else if (!types.Contains(ct)) { valid = false; reason = $"{cid}: change_type not allowed"; }
            else if (!risks.Contains(rl)) { valid = false; reason = $"{cid}: risk_level not valid"; }
            else { valid = true; reason = $"{cid}: pre-validation passed for {svc}"; }

            var res = new JsonObject { ["action"] = valid ? "complete" : "fail", ["valid"] = valid, ["reason"] = reason };
            await client.ResumeIntentAsync(id, res, new RequestOptions { OwnerAgent = addr });
            Console.WriteLine($"resumed {id} (valid={valid})");
        }
    }
    catch { await Task.Delay(2000); }
}
